### PR TITLE
Unify consistency tests for mbedtls_pk_import_into_psa and mbedtls_pk_copy_from_psa

### LINKED ISSUE #8836: Unify consistency tests for mbedtls_pk_import_into_psa and mbedtls_pk_copy_from_psa
We should do pair consistency testing on `mbedtls_pk_import_into_psa`, i.e. ensure that the imported key has the expected material.

For key pairs, since we also have a pk object, we can do some pair consistency checks: sign with one key and verify with the other, or encrypt/decrypt, or do a key agreement. Valerio has done this for `mbedtls_pk_copy_from_psa` in https://github.com/Mbed-TLS/mbedtls/pull/8774.

For either key pairs or public keys, we can check the consistency by comparing the exported public keys in a canonical format. Gilles has done this for `mbedtls_pk_import_into_psa` in #8807.

The goal of this issue is to unify the consistency testing. Presumably using the export-public method since it's a lot faster.

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

Resolves #8836.

## PR checklist

- [x] **changelog** not required - tests only
- [x] **backport** not required - tests for functions that are not in 2.28
- [x] **tests** provided (the PR is all about testing)

### CODE REVIEW

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

File: tests/suites/test_suite_pk.function [hunk 1]
```diff
@@ -2622,135 +2616,23 @@ void pk_copy_from_psa_success(data_t *priv_key_data, int key_type_arg,
...
+    /* Check consistency between copied PSA keys and generated PK contexts. */
+    TEST_EQUAL(mbedtls_test_key_consistency_psa_pk(priv_key_id, &pk_priv), 1);
+    TEST_EQUAL(mbedtls_test_key_consistency_psa_pk(priv_key_id, &pk_pub), 1);
+    TEST_EQUAL(mbedtls_test_key_consistency_psa_pk(pub_key_id, &pk_priv), 1);
+    TEST_EQUAL(mbedtls_test_key_consistency_psa_pk(pub_key_id, &pk_pub), 1);
 
     /* Test that the keys from mbedtls_pk_copy_public_from_psa() are identical
-     * to the public key from mbedtls_pk_copy_from_psa(). */
+     * to the public keys from mbedtls_pk_copy_from_psa(). */
     mbedtls_test_set_step(1);
```
Comment (mpg) [inline tests/suites/test_suite_pk.function:2630]: Pre-existing:  I don't think calling `set_step()` here is useful - on tests failures the line number of the failure will be printed which is enough. IMO `set_step()` is only useful for loop, where the line number is not enough: you want to know in which iteration the failure happened.

(OTOH, not harmful either.)

**Review by mpg — APPROVED** [review]
LGTM, thanks!

Commented on a pre-existing point, but not worth doing another round for that, just saying for the future.

File: tests/suites/test_suite_pk.function (hunk 1 above)
Comment (gilles-peskine-arm) [reply tests/suites/test_suite_pk.function:2630]: `set_step` is also useful before calling auxiliary functions that can report failures. In this case, setting the step allows distinguishing between failures inside the first call to `pk_public_same` vs the second call.

**Pull request merged**